src/rgw/rgw_s3select.cc

```diff
@@ -37,6 +38,7 @@ uint64_t aws_response_handler::get_total_bytes_returned()
 void aws_response_handler::update_total_bytes_returned(uint64_t value)
 {
   total_bytes_returned = value;
+  s->s3select_usage.bytes_returned = total_bytes_returned;
```

Comment:
:+1: 

as a note for myself, the parquet flow is currently not integrated with `aws_response_handler`

Reply:
@galsalomon66 Seems parquet object is also inheriting from `base_s3object` (https://github.com/ceph/s3select/blob/master/include/s3select.h#L2804) - and the same function for `result_values_to_string` is being called for all engines. Can't we just update `m_aws_response_handler` by the value of `get_return_result_size` from the parquet object?
and also for processed value just update it from the iterator?
Something like this:
```diff
diff --git a/src/rgw/rgw_s3select.cc b/src/rgw/rgw_s3select.cc
index c7eaa69842f..696b765675b 100644
--- a/src/rgw/rgw_s3select.cc
+++ b/src/rgw/rgw_s3select.cc
@@ -484,6 +484,7 @@ int RGWSelectObj_ObjStore_S3::run_s3select_on_parquet(const char* query)
     fp_result_header_format(m_aws_response_handler.get_sql_result());
     //at this stage the Parquet-processing "takes control", it keep calling to s3-range-request according to the SQL statement.
     status = m_s3_parquet_object.run_s3select_on_object(m_aws_response_handler.get_sql_result(), fp_s3select_result_format, fp_result_header_format);
+    m_aws_response_handler.update_total_bytes_returned( m_s3_parquet_object.get_return_result_size() );
     if (status < 0) {
       m_aws_response_handler.get_sql_result().append(m_s3_parquet_object.get_error_description());
       fp_s3select_result_format(m_aws_response_handler.get_sql_result());
@@ -761,6 +762,7 @@ int RGWSelectObj_ObjStore_S3::parquet_processing(bufferlist& bl, off_t ofs, off_
       requested_buffer.append(&(it)[0]+ofs, len);
     }
     ldout(s->cct, 10) << "S3select:append_in_callback = " << append_in_callback << dendl;
+    m_aws_response_handler.update_processed_size(append_in_callback);
     if (requested_buffer.size() < m_request_range) {
       ldout(s->cct, 10) << "S3select: need another round buffe-size: " << requested_buffer.size() << " request range length:" << m_request_range << dendl;
       return 0;
```

Reply:
Hi @clwluvw,

yes, the `result_values_to_string` is shared by the different readers (CSV, JSON , and Parquet)
by design, the RGW and s3select are decoupled (s3select-engine gets bytes to process and it returns results)

the Parquet processing differs from other processing since it "leads" the object scanning (in contrary to CSV and JSON)
the engine decides which bytes-ranges should be scanned.

the `fp_s3select_result_format` gets the result(final) and "prepares" to send it back to the client.
at that point, we can count the `return bytes` 

the `range_request` method gets *all* the requests of Parquet-engine for the scanned-bytes, thus we can count the `scanned bytes` at that point.

the `range_request` and `fp_s3select_result_format` are defined in RGW.

we can merge this PR and later we can provide the necessary changes for Parquet flow.
(its not related to this change, but we need to upgrade the arrow libraries)